Feature/unnecessary sunders

johnnynaps opened the pull request.

johnnynaps committed:
added unnecessarySunder tracker and a count to the formatted table.

johnnynaps committed:
created ordered maps of unnec sundered mobs sorted by times sundered

johnnynaps committed:
Created a method to sort and format mob names and values into a string.

Prints that formatted string in the previously formatted table

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/CombatLogAnalyzer.java:
```diff
@@ -125,9 +136,49 @@ private boolean lineIsASuccessfulSunder(String[] lineAsArray) {
+     * Formats a string representation of the top 3 creatures and their associated
+     * values.
+     * 
+     * @param unnecessayCreatureList
```
Please add a parameter description for what this parameter is.
Also, change spelling of unnecessay --> unnecessary

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/CombatLogAnalyzer.java:
```diff
@@ -125,9 +136,49 @@ private boolean lineIsASuccessfulSunder(String[] lineAsArray) {
+     * 
+     * @param unnecessayCreatureList
+     * @return formatted string representation of top 3 creature names and
+     *         associated values
```
Please describe what the actual values are.

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/CombatLogAnalyzer.java:
```diff
@@ -125,9 +136,49 @@ private boolean lineIsASuccessfulSunder(String[] lineAsArray) {
+        Map<String, Integer> unnecessaryCreatureMapOrdered = new LinkedHashMap<>();
+        String topSunderMobsString = "";
+
+        for (int i = 0; i < unnecessayCreatureList.size(); i++) {
```
Could be...


```Java
for (Creature creature : unnecessaryCreatureList) {

}
```

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/CombatLogAnalyzer.java:
```diff
@@ -125,9 +136,49 @@ private boolean lineIsASuccessfulSunder(String[] lineAsArray) {
+                unnecessaryCreatureMapUnordered.put(creature.getName(), 1);
+            }
+        }
+
```
A single line comment here explaining what the stream is doing would be beneficial.

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/CombatLogAnalyzer.java:
```diff
@@ -125,9 +136,49 @@ private boolean lineIsASuccessfulSunder(String[] lineAsArray) {
+                        unnecessaryCreatureMapUnordered.get(creature.getName()) + 1);
+            }
+            else {
+
```
Unnecessary blank line 😃

rprescottjr commented on src/main/java/com/rprescott/combatloganalyzer/services/SunderTracker.java:
```diff
@@ -41,31 +42,50 @@ public void insertSunder(String[] combatLogLineAsArray) {
     }
-    
+
+    public Map<String, List<Creature>> getUnnecessarySunders(List<String> mobNames) {
```
Please provide a Javadoc of what this method does.

johnnynaps committed:
Addressed requested changes

added javadoc comments to SunderTracker and CombatLogAnalyzer. Changed
general styling of sections.
